src/XMakeBuildEngine/BackEnd/TaskExecutionHost/TaskExecutionHost.cs

```diff
@@ -185,7 +185,7 @@ internal TaskExecutionHost()
         /// </summary>
         ~TaskExecutionHost()
         {
-            Debug.Fail("Unexpected finalization.  Dispose should already have been called.");
+            // Debug.Fail("Unexpected finalization.  Dispose should already have been called.");
```

Comment:
More details in https://github.com/rainersigwald/msbuild/commit/ea574a45c6856127c38957716e446b3a150687fe and #234 but it looks like a) we're not reliably running `Dispose` and b) throwing an exception from a finalizer crashes in CoreCLR.